packages/amazonq/src/auth/util.ts

```diff
@@ -28,10 +29,14 @@ export async function getAuthStatus(): Promise<AuthStatus> {
         // Determine the status of the Toolkit connection we will autoconnect to
         if (autoConnectConn) {
             authStatus = autoConnectConn.state === 'valid' ? 'connected' : 'expired'
+
+            // Though TS won't say it, AwsConnection sufficiently overlaps with Connection for the purposes
+            // of `getAuthFormIdsFromConnection`
+            authEnabledConnections = AuthUtils.getAuthFormIdsFromConnection(autoConnectConn as unknown as Connection)
```

Comment:
future / not a blocker: would `Partial<>` work instead?

Reply:
Yes, but we would need to adjust the types of a bunch of utility functions :(